src/app/shared/form/fancy-field.component.ts

```diff
@@ -15,15 +17,36 @@ export class FancyFieldComponent {
   @Input() name: string;
   @Input() changed: string;
   @Input() invalid: string;
-
-  // Form field options
-  @Input() textinput: boolean;
-  @Input() textarea: boolean;
-  @Input() select: string[];
   @Input() label: string;
   @Input() invalidlabel: string;
-  @Input() small: boolean;
-  @Input() required: boolean;
+  @Input() hideinvalid: boolean;
+
+  // Form field types (intercepted with defaults)
+  type: string;
+  _select: string[];
+  @Input()
+  set textinput(any: any) { this.type = 'textinput'; }
+  @Input()
+  set number(any: any) { this.type = 'number'; }
+  @Input()
+  set textarea(any: any) { this.type = 'textarea'; }
+  @Input()
+  set select(opts: string[]) { this.type = 'select'; this._select = opts || []; }
+  get select() { return this._select; }
+
+  // Field attributes
+  _small = false;
+  _inline = false;
+  _required = null;
+  @Input()
+  set small(small: boolean) { this._small = isset(small); }
```

Comment:
These interceptors allow a simpler syntax for using this component ... instead of doing `small="true"` in the template, you can just use the standalone attribute `small`.